pf/tfbridge/provider_check.go

```diff
@@ -48,7 +48,8 @@ func (p *provider) CheckWithContext(
 
 	if info := rh.pulumiResourceInfo; info != nil {
 		if check := info.PreCheckCallback; check != nil {
-			checkedInputs, err := check(ctx, checkedInputs, p.lastKnownProviderConfig.Copy())
+			var err error
+			checkedInputs, err = check(ctx, checkedInputs, p.lastKnownProviderConfig.Copy())
```

Comment:
Shadowing considerer harmful.. it got me here again.

Reply:
I think the pain point is that the difference between a declaration and a mutation is `:`. It somehow bothers me less then its `let ` or `auto ` or `(let ((`.